pkg/agent/agentserver/server.go

```diff
@@ -196,6 +215,68 @@ func (s *ProxyServer) serveSend(stream agent.ProxyService_ProxyServer, sendCh <-
 	}
 }
 
+func (s *ProxyServer) validateAuthToken(token string) error {
+	trReq := &authv1.TokenReview{
+		Spec: authv1.TokenReviewSpec{
+			Token:     token,
+			Audiences: []string{s.AgentAuthenticationOptions.AuthenticationAudience},
```

Comment:
You mentioned that Audiences is mandatory, is it something standardized across other system pods? I noticed that it's optional in the api (https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/authentication/v1/types.go#L73) and didn't see any other instances of it being used in k/k.

Reply:
cc @mikedanese

Reply:
Since bearer tokens can be replayed by the relying party, security is predicated on the assumption that a single bearer token is only valid against the relying party that they it was originally intended for.

> To deal with token redirect, it is important for the authorization
> server to include the identity of the intended recipients (the
> audience), typically a single resource server (or a list of resource
> servers), in the token.

https://tools.ietf.org/html/rfc6750#section-5.2

All uses of tokens in k/k are to communicate with the kube-apiserver.  The audience in a TokenRequest and TokenReview defaults to the kube-apiserver. Since the konnectivity server is not the kube-apiserver, it should not accept tokens that are audience bound the kube-apiserver (just like the kube-apiserver should not accept tokens that are audience bound to the konnectivity server).

Reply:
That makes sense. Thanks!